checkov/common/graph/graph_builder/graph_components/blocks.py

```diff
@@ -110,6 +110,14 @@ def get_attribute_dict(self, add_hash: bool = True) -> Dict[str, Any]:
             # since module names are user defined we are just setting 'module' as resource type for easier searching
             base_attributes[CustomAttributes.RESOURCE_TYPE] = "module"
 
+        if self.block_type == BlockType.PROVIDER:
+            provider_name = base_attributes[CustomAttributes.BLOCK_NAME]
+            if isinstance(provider_name, str):
+                provider_type = provider_name.split(".")[0]
```

Comment:
I split the string to take the provider type because the block name is the "providerType.providerAliases" and we need to match it with resource_types in the check 
ex: block_name = aws.example1
resource_type = provider.aws